Tests/UnitTests/Sources/GREYActionsTest.m

```diff
@@ -198,5 +198,35 @@ - (void)testInvalidTapActionSucceedsAfterDisablingConstraints {
   }
 }
 
+- (void)testReplaceTextSucceedsOnUITextField {
+    NSString *textToReplace = @"A String";
+    NSError *error;
+
+    UITextField *textfield = [[UITextField alloc] init];
+
+    id<GREYAction> replace = [GREYActions actionForReplaceText:textToReplace];
+    [replace perform:textfield error:&error];
+
+    XCTAssertNil(error);
+
+    XCTAssertEqualObjects(textToReplace, textfield.text);
+}
+
+- (void)testReplaceTextFailsOnUIView {
+    NSString *textToReplace = @"A String";
+    NSError *error;
+
+    UIView *label = [[UIView alloc] init];
+
+    id<GREYAction> replace = [GREYActions actionForReplaceText:textToReplace];
+    [replace perform:label error:&error];
+
+    NSString *reasonSubstring = @"Action could not be performed on element";
```

Comment:
> If the element doesn't support it then show that.

If the element doesn't support it then we'd get a constraint failed message, right? 

```swift
constraints = grey_anyOf(grey_respondsToSelector(@selector(text)),
                         grey_kindOfClass(webElement),
                         nil);
```

> We could also do an assert possibly to ensure that the element has been cleared for sure.

We aren't clearing the element though, just setting the text attribute. The existing APIs in EarlGrey don't work well for asserting if the elements text value has updated. For example, it'll fail completely in a WebView. I think we only want grey_replaceText to do one thing: replace the text.

> If the setter didn't succeed then that.

That seems different from how we're doing type text?

> Usually, we have the action that checks for the failure because it is granular enough.

I'll look into improving the failure handling.